lib/ziti-tunnel-cbs/ziti_tunnel_ctrl.c

```diff
@@ -802,20 +803,22 @@ static void on_enable_mfa(ziti_context ztx, int status, ziti_mfa_enrollment *enr
 
     struct ziti_instance_s *inst = ziti_app_ctx(ztx);
     mfa_event *ev = calloc(1, sizeof(struct mfa_event_s));
-    ev->operation = strdup(mfa_status_name(mfa_status_enrollment_challenge));
+    ev->operation = calloc(strlen(mfa_status_name(mfa_status_enrollment_challenge)), sizeof(char));
+    ev->operation = mfa_status_name(mfa_status_enrollment_challenge);
     ev->operation_type = mfa_status_enrollment_challenge;
+    ev->provisioning_url = calloc(strlen(enrollment->provisioning_url), sizeof(char));
     ev->provisioning_url = strdup(enrollment->provisioning_url);
     char **rc = enrollment->recovery_codes;
-    int code_len = 0;
+    int size = 0;
     while (*rc != NULL) {
-        code_len = code_len + strlen(*rc);
         rc++;
+        size++;
     }
-    ev->recovery_codes = malloc(code_len + 1);
+    ev->recovery_codes = malloc((size + 1) * sizeof(char *));
     int idx;
     for (idx=0; enrollment->recovery_codes[idx] !=0; idx++) {
         ev->recovery_codes[idx] = calloc(strlen(enrollment->recovery_codes[idx]), sizeof(char));
```

Comment:
this is leaked. the pointer returned by calloc is overwritten by the pointer returned by strdup.